### ISSUE TITLE
Entry attributeLabel for 'parent' for custom validator

### ISSUE THREAD

**Issue by jerome2710** [craftcms/cms#6007]
### Description
Would it be possible to add 'parent' as an attributeLabel to the Entry-class?

I am adding a rule for specific entry types and adding a custom validator on the parent so this entry type cannot be nested. However, Yii's `addError` fails because there is no attributeLabel for 'parent'.

```
Event::on(
    Entry::class,
    Entry::EVENT_DEFINE_RULES,
    static function(DefineRulesEvent $event) {
        /** @var Entry $entry */
        $entry = $event->sender;

        if ($entry->getSection()->handle === 'sectionHandle' && $entry->getType()->handle === 'typeHandle') {
            foreach(SomeRuleClass::define() as $rule) {
                $event->rules[] = $rule;
            }
        }
    }
);
```

```
class SomeRuleClass
{
    public static function define(): array
    {
        return [
            [
                'parent',
                EmptyParentValidator::class,
            ]
        ];
    }
}
```

```
class EmptyParentValidator extends Validator
{
    public function validateValue($value)
    {
        if (!empty($value)) {
            return 'Parent should be empty';
        }

        return null;
    }
}
```

### Additional info

- Craft version: 3.4.17.1
- PHP version: 7.2.4
- Database driver & version: MySQL 5.5.64

**Comment by brandonkelly** [comment]
Yii will generate its own attribute label based on the attribute name, if a label is not explicitly provided:

https://github.com/yiisoft/yii2/blob/60a39654dd2d425a24c64d6c1d641a0754b015bc/framework/base/Model.php#L535

https://github.com/yiisoft/yii2/blob/60a39654dd2d425a24c64d6c1d641a0754b015bc/framework/base/Model.php#L699-L702

Are you getting an error of some sort that’s not expected?

**Comment by jerome2710** [comment]
Hi @brandonkelly,

Thank you for looking into this. The error produced by `validateValue` in my custom `EmptyParentValidator` is added to the field by `addError()` in Yii's `yii\validators\Validator::validateAttribute`:

https://github.com/yiisoft/yii2/blob/master/framework/validators/Validator.php#L305

I was getting the error `Illegal string offset 'attribute'`, which I figured was caused by the missing attributeLabel.

https://github.com/yiisoft/yii2/blob/master/framework/validators/Validator.php#L442

As you pointed out it should be auto-generated, so I did some extra digging. It's not actually the label giving problems, but the error itself. The return value of `validateValue` should be an array with messages => parameters, not just the message as a string. Adjusted that and it solved the problem.
